### PR TITLE
detect-flowbits: add details for flowbits v7

### PR DESCRIPTION
**Pull request by hadiqaalamdar**
Task #6309

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/6309) ticket: https://redmine.openinfosecfoundation.org/issues/6309
Previous PR: https://github.com/OISF/suricata/pull/10008

Describe changes:
- added the recommended changes from the last PR. The operator string is working correctly for or operator
- did get rebase appropriately.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1529

### CODE REVIEW

File: src/detect-engine-analyzer.c [hunk 1]
```diff
@@ -861,6 +863,51 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
...
+                        break;
+                    case DETECT_FLOWBITS_CMD_UNSET:
+                        jb_set_string(js, "cmd", "unset");
+                        break;
+                    case DETECT_FLOWBITS_CMD_TOGGLE:
+                        jb_set_string(js, "cmd", "toggle");
+                        break;
+                }
+                int flag = 0;
+                if (cd->cmd != DETECT_FLOWBITS_CMD_NOALERT) {
```
[the remark this answers is not in the document]
Comment (inashivb) [reply src/detect-engine-analyzer.c:891]: I don't think we do. I think we should also remove this constant though bc it gives me an impression that this is a valid flowbit specific command and while it is used w flowbits, it is never really used to set `cmd`.

File: src/detect-engine-analyzer.c [hunk 2]
```diff
@@ -861,6 +863,51 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
...
+                        jb_set_string(js, "cmd", "set");
+                        break;
+                    case DETECT_FLOWBITS_CMD_UNSET:
+                        jb_set_string(js, "cmd", "unset");
+                        break;
+                    case DETECT_FLOWBITS_CMD_TOGGLE:
+                        jb_set_string(js, "cmd", "toggle");
+                        break;
+                }
+                int flag = 0;
```
[the remark this answers is not in the document]
Comment (inashivb) [reply src/detect-engine-analyzer.c:890]: +1 to everything

File: src/detect-engine-analyzer.c [hunk 3]
```diff
@@ -861,6 +863,51 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
...
+                            const char *varname =
+                                    VarNameStoreSetupLookup(cd->or_list[i], VAR_TYPE_FLOW_BIT);
+                            jb_append_string(js, varname);
+                        }
+                    }
+                    jb_close(js); // array
+                    if (flag == 1) {
+                        jb_set_string(js, "operator", "or");
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (inashivb) [reply src/detect-engine-analyzer.c:907]: Indeed. We don't define it as a specific operator anyway like we do for or. It's only fair. 😛

**Comment by inashivb** [comment]
> Kudos for the progress! :)
> 
> I'm still considering what should be the approach for the `noalert` case - might be we're missing documentation, maybe it's a case where we need a warning - like we have for the flowbits that haven't been set...
> 

I think it is fine as-is. We do mention in our docs that it is identical to standalone `noalert;`: https://docs.suricata.io/en/latest/rules/differences-from-snort.html#flowbits-noalert
If you'd also want to have this in flowbits section of docs, I'll support that. :)

File: src/detect-engine-analyzer.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (inashivb) [reply src/detect-engine-analyzer.c:891]: wrt I have made https://github.com/OISF/suricata/pull/10021
If it gets accepted, please rebase your code on that PR and remove all usages of `DETECT_FLOWBITS_CMD_NOALERT`.
If it gets rejected, maybe just add some notes like Juliana mentioned about noalert control never getting to this part.

**Comment by hadiqaalamdar** [comment]
New PR: https://github.com/OISF/suricata/pull/10026

**Pull request closed without merging**